app/src/main/java/com/example/saar/Login_SignUp/SignUpFragment.java

```diff
@@ -1,21 +1,118 @@
+import android.widget.EditText;
+import android.widget.Spinner;
 
 import com.example.saar.R;
 
+import java.util.Calendar;
+
 public class SignUpFragment extends Fragment {
 
+    EditText edittext;
+    DatePickerDialog.OnDateSetListener setListener;
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_sign_up, container, false);
+        View rootView = inflater.inflate(R.layout.fragment_sign_up, container, false);
+
+        Spinner spinnerGraduationYear = (Spinner) rootView.findViewById(R.id.spinner_graduation_year);
+        Spinner spinnerDegree = (Spinner) rootView.findViewById(R.id.spinner_degree);
+        Spinner spinnerDepartment = (Spinner) rootView.findViewById(R.id.spinner_department);
+        Spinner spinnerEmploymentType = (Spinner) rootView.findViewById(R.id.spinner_employment_type);
+
+        // Initializing ArrayAdapter
+        ArrayAdapter<CharSequence> spinnerGraduationYearArrayAdapter = ArrayAdapter.createFromResource(getContext(), R.array.graduation_year_array, android.R.layout.simple_spinner_item);
+        ArrayAdapter<CharSequence> spinnerDegreeArrayAdapter = ArrayAdapter.createFromResource(getContext(), R.array.degree_array, android.R.layout.simple_spinner_dropdown_item);
+        ArrayAdapter<CharSequence> spinnerDepartmentArrayAdapter = ArrayAdapter.createFromResource(getContext(), R.array.department_array, android.R.layout.simple_spinner_item);
+        ArrayAdapter<CharSequence> spinnerEmploymentTypeArrayAdapter = ArrayAdapter.createFromResource(getContext(), R.array.employment_type_array, android.R.layout.simple_spinner_item);
+
+        spinnerGraduationYearArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinnerDegreeArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinnerDepartmentArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinnerEmploymentTypeArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        spinnerGraduationYear.setAdapter(spinnerGraduationYearArrayAdapter);
+        spinnerDegree.setAdapter(spinnerDegreeArrayAdapter);
+        spinnerDepartment.setAdapter(spinnerDepartmentArrayAdapter);
+        spinnerEmploymentType.setAdapter(spinnerEmploymentTypeArrayAdapter);
+
+        spinnerGraduationYear.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+            }
+        });
+        spinnerDegree.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
```

Comment:
every listener should be placed inside oniewCreated, else the app will crash during extortion of data. Error message will be null pointer reference. Please make the changes. @somenath1435 suggest?